File: web/src/lib/types/product.ts
/**
 * Product response shape returned by the C# API (GET /api/v1/product).
 * Mirrors the api Dtos/ProductResponse. Request validation lives server-side
 * (FluentValidation), so the frontend only needs the response type.
 */
export interface ProductResponse {
  id: number;
  name: string;
  price: number;
  sectionId: number;
}

File: api/Domain/Product.cs
namespace Api.Domain;

/// <summary>A product belonging to a section. Mirrors the TS `products` table.</summary>
public class Product
{
    public int Id { get; set; }
    public string Name { get; set; } = string.Empty;
    public double Price { get; set; }
    public int SectionId { get; set; }

    public Section? Section { get; set; }
}
